### DISCUSSION TITLE
Reset Notebook State

### DISCUSSION THREAD

**Discussion by rpena-anaconda** [jupyterlite/jupyterlite#322] [Q&A]
Sorry in advance if this is a dumb question! 😄 

I was hoping to see if anyone has any insight on _how to reset the state of a Jupyter Lite notebook to its original state_ (pre user modifying cells, etc.). Say I have a notebook running, with some fields populated setting up matplotlib and a bikeshare data set. I go in and start fiddling with the dataset looking at other insights but now, I can't go back to my original state.

I had two ideas but not sure on execution/direction.

1. IndexedDB stores the state. In my case, we iframe JupyterLite notebook so because of same origin policy, we aren't allowed to delete DB.
2. Is there a way to "hijack" index.html file with my own? My index.html would have some JS snippet to implement idea #1. I'd be on same origin so 🤞  it may work to delete DB.

**Comment by jtpio** [comment]
Thanks @rpena-anaconda.

> Say I have a notebook running, with some fields populated setting up matplotlib and a bikeshare data set. I go in and start fiddling with the dataset looking at other insights but now, I can't go back to my original state.

Was the notebook pre-populated as static content? Using https://jupyterlite.readthedocs.io/en/latest/configuring.html#adding-content

If so then it's possible to delete the local copy stored in IndexedDB, and the notebook will revert back to the original deployed content:

https://user-images.githubusercontent.com/591645/132943267-9c86656c-4e20-406b-94b3-2728129a3930.mp4

**Comment by rpena-anaconda** [comment]
@jtpio Thanks for getting back to me! 

We are serving it as a static asset. We are using the **_RetroLite_** theme for the experience. I think the conundrum is that we are drilled into the notebook versus the tree view like in the example above. I added image as an example.

<img width="1461" alt="Screen Shot 2021-09-14 at 1 00 17 PM" src="https://user-images.githubusercontent.com/48453272/133310336-4ba44e62-c5fd-4599-98d0-45ce914e0d11.png">

We did it similar to the RetroLab example on the [Jupterlite homepage](https://jupyterlite.readthedocs.io/en/latest/). Initially you see the file system, double clicking on a notebook takes you to the notebook view.

I hope that makes sense.

**Comment by jtpio** [comment]
Ah thanks @rpena-anaconda for sharing the screenshot.

So maybe we need something like https://github.com/jupyterlite/jupyterlite/issues/9? That would be available as a command and item menu, so it also works in Retro.

**Comment by rpena-anaconda** [comment]
@jtpio Yup, #9 that pretty much captured it! Thanks again for your help! Greatly appreciated it!
